README.markdown

```diff
@@ -125,170 +70,157 @@ You can also use this parameter to set a specific registry value as the default
 
-###Defined Type: registry::service
+####`registry_key`
 
-The `registry::service` defined resource type utilizes specific Registry keys and values to manage service entries in the Microsoft service control framework. Specifically, `registry::service` manages the values in the key HKLM\System\CurrentControlSet\Services\$name\.
+#####`ensure`
 
-    registry::service { puppet:
-      ensure       => present,
-      display_name => "Puppet Agent",
-      description  => "Periodically fetches and applies configurations from a Puppet master server.",
-      command      => 'C:\PuppetLabs\Puppet\service\daemon.bat',
-    }
+*Optional.* Tells Puppet whether the key should or shouldn't exist. Valid options: 'present' and 'absent'. Default value: if unspecified, Puppet won't enforce any other parameters unless the key is already present on a node.
 
-This is an alternative approach to using INSTSRV.EXE [1](http://support.microsoft.com/kb/137890).
+#####`path`
 
-####Parameters in `registry::service`:
+*Required.* Specifies a Registry key for Puppet to manage. If any of the parent keys in the path don't exist, Puppet creates them automatically. Valid options: a string containing a Registry path. For example: 'HKLM\Software' or 'HKEY_LOCAL_MACHINE\Software\Vendor'.
 
-#####`ensure`
+If Puppet is running on a 64-bit system, the 32-bit Registry key can be explicitly managed using a prefix. For example: '32:HKLM\Software'.
 
-Determines whether or not the service is 'present' or 'absent'.
+#####`purge_values`
 
-#####`display_name`
+*Optional.* Specifies whether to delete any values in the specified key that are not managed by Puppet. Valid options: 'true' and 'false'. Default value: 'false'.
 
-Sets the Display Name of the service.  Defaults to the title of the resource.
+For more on this parameter, see the [Purge existing values section](#purge-existing-values) under Usage.
 
-#####`description`
+####`registry_value`
 
-A description of the service.
+#####`path`
 
-#####`command`
+*Optional.* Specifies a Registry value for Puppet to manage. Valid options: a string containing a Registry path. If any of the parent keys in the path don't exist, Puppet creates them automatically. For example: 'HKLM\Software' or 'HKEY_LOCAL_MACHINE\Software\Vendor'. Default value: the title of your declared resource.
 
-Specifies the command to execute.
+If Puppet is running on a 64-bit system, the 32-bit Registry key can be explicitly managed using a prefix. For example: '32:HKLM\Software\Value3'.
 
-#####`start`
+#####`ensure`
 
-Specifies the starting mode of the service. Valid values are 'automatic', 'manual', and 'disabled'. 
+*Optional.* Tells Puppet whether the value should or shouldn't exist. Valid options: 'present' and 'absent'. Default value: unspecified (if left unspecified, Puppet won't enforce any other parameters unless the value is already present on a node.)
```

Comment:
Updated copy: 

##### `ensure`

Tells Puppet whether the value should or shouldn't exist. Valid options: 'present' and 'absent'. Default value: 'present'.